velox/common/memory/Memory.h

```diff
@@ -67,87 +67,90 @@ constexpr uint16_t kDefaultAlignment = 64;
 
 class ScopedMemoryPool;
 
-class MemoryPool {
+class MemoryPool : public std::enable_shared_from_this<MemoryPool> {
  public:
-  virtual ~MemoryPool() {}
+  MemoryPool(const std::string& name, std::shared_ptr<MemoryPool> parent);
+  virtual ~MemoryPool();
+
+  ////////////////////////// tree methods ////////////////////////////
```

Comment:
> tree methods

This is not very descriptive. Would you update?

Reply:
Updated.